lib/index.js

```diff
@@ -533,15 +533,24 @@ internals.sign = async function (name, value, options) {
 
 exports.exclude = function (cookies, excludes) {
 
-    let result = '';
-    const verify = cookies.replace(internals.pairsRx, ($0, $1, $2) => {
+    const result = [];
 
-        if (excludes.indexOf($1) === -1) {
-            result = result + (result ? ';' : '') + $1 + '=' + $2;
+    const chunks = cookies.split(';');
+
+    for (const chunk of chunks) {
+        const match = internals.pairsRx.exec(chunk.trim());
+
+        if (!match) {
+            return Boom.badRequest('Invalid cookie header');
         }
 
-        return '';
-    });
+        const name = match[1];
+        const value = match[2];
+
+        if (excludes.indexOf(name) === -1) {
+            result.push(`${name}=${value}`);
+        }
+    }
 
-    return verify === '' ? result : Boom.badRequest('Invalid cookie header');
+    return result.join(';');
```

Comment:
This should probably be `join('; ')`, with an extra space, since this is [the proper syntax](https://www.rfc-editor.org/rfc/rfc6265#section-4.2.1).